packages/cli/src/cmds/stats/stats.ts

```diff
@@ -1,335 +1,51 @@
-
-            console.log(
-              chalk.underline(
-                [
-                  'Time'.padStart(9),
-                  '%'.padStart(5),
-                  'Count'.padStart(7),
-                  'Function name'.padEnd(30),
-                ].join(' | ')
-              )
-            );
-            slowestExecutionTimes.forEach((executionTime) => {
-              const displayMs = [
-                Math.round(executionTime.elapsed_instrumentation_time_total * 1000).toString(),
-                'ms',
-              ].join('');
-              const displayPercent = `${(
-                Math.round(
-                  (executionTime.elapsed_instrumentation_time_total / totalInstrumentationTime) *
-                    1000
-                ) / 10
-              ).toLocaleString('en-us', { maximumFractionDigits: 2, minimumFractionDigits: 0 })}%`;
-
-              console.log(
-                [
-                  displayMs.padStart(9),
-                  displayPercent.padStart(5),
-                  executionTime.num_calls.toString().padStart(7),
-                  executionTime.name.split(':').slice(1).join(':'),
-                ].join(' | ')
-              );
-            });
-          }
-        }
-
-        let telemetryMetrics = {};
-        let telemetryMetricsCounter = 1;
-        biggestAppMapSizes.forEach((appmap) => {
-          telemetryMetrics[`biggestAppmaps_${telemetryMetricsCounter}`] = appmap.size;
-          telemetryMetricsCounter += 1;
-        });
-        telemetryMetricsCounter = 1;
-        slowestExecutionTimes.forEach((executionTime) => {
-          telemetryMetrics[`slowestInstrumentationTimesTotal_${telemetryMetricsCounter}`] =
-            executionTime.elapsed_instrumentation_time_total;
-          telemetryMetrics[`slowestInstrumentationTimesPath_${telemetryMetricsCounter}`] =
-            executionTime.path;
-          telemetryMetricsCounter += 1;
-        });
-
-        Telemetry.sendEvent({
-          name: `stats:${handlerCaller}:success`,
-          properties: {
-            path: appMapDir,
-          },
-          metrics: telemetryMetrics,
-        });
-      } catch (err) {
-        let errorMessage: string | undefined = (err as any).toString();
-        if (err instanceof Error) {
-          Telemetry.sendEvent({
-            name: `stats:${handlerCaller}:error`,
-            properties: {
-              errorMessage,
-              errorStack: err.stack,
-            },
-          });
-        }
-      }
-
-      return [biggestAppMapSizes, slowestExecutionTimes];
-    }
-
-    return await showStats();
-  };
-
-  return await commandFn();
+export async function handler(
+  argv: any,
+  handlerCaller: string = 'from_stats'
+): Promise<Array<EventInfo> | Promise<[SortedAppMapSize[], SlowestExecutionTime[]]>> {
+  if (argv.f) {
+    return await statsForMap(argv);
+  }
+  return await statsForDirectory(argv, handlerCaller);
```

Comment:
Without either `--appmap-file` or `--appmap-dir`, the command should analyze `appmap_dir`.